LibGAP.jl/src/ccalls.jl

```diff
@@ -85,6 +85,20 @@ function propertynames(funcobj::GlobalsType,private)
     return tuple(list_converted...)
 end
 
+function to_gap(obj::Any)
+    ret_ptr = ccall(:_ConvertedFromJulia_internal,Ptr{Cvoid},(Any,),obj)
+    ret_val = GET_FROM_GAP(ret_ptr)
+    return ret_val
+end
+
+
+function from_gap(obj::MPtr,::Any)
+    ret_val = ccall(:_ConvertedToJulia_internal,Any,(MPtr,),obj)
```

Comment:
So I guess the idea is to have this two methods as kind of the "default" fallback methods, and implement any custom user driven conversions in Julia code?

That is already an improvement, but of course still asymmetric. E.g. it would now be natural to implement the conversion of GAP bigints to Julia bigints in Julia; but then it would not be accessible in JuliaInterface... so I guess we'd also need to change `JuliaInterface` to call `to_gap` / `from_gap` in certain cases (but of course from a higher level function, not from `_ConvertedFromJulia_internal`, to avoid infinite recursion)

Reply:
Yep, indeed. I think the ultimate goal needs to be exactly one conversion function for each type pair in GAP/Julia, either written on Julia side, so no code is actually duplicated.